### PR TITLE
 MM-21319 : Migrate "/integration/enterprise/ldap_group" e2e tests to TypeScript

### PR DESCRIPTION
**Pull request by cyberbuff**
#### Summary

Converting following spec files under /integration/enterprise/ldap_group to Typescript

1. channel_modes_spec.js
2. groups_assign_roles_spec.js
3. search_channels_spec.js
4. group_mentions_spec.js
5. invite_botspec.js
6. team_and_channel_assign_rolesspec.js

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/21319

#### Related Pull Requests

No related PRs. 

#### Screenshots

No UI changes.

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by josephbaylon — APPROVED** [review]
LGTM. Thanks @cyberbuff !!

File: e2e/cypress/tests/integration/enterprise/ldap_group/invite_bot_spec.ts [hunk 1]
```diff
@@ -43,7 +49,12 @@ describe('Group Synced Team - Bot invitation flow', () => {
 
         // # Logout sysadmin and login as an LDAP Group synced user
         cy.apiLogout();
-        cy.apiLogin({username: 'test.one', password: 'Password1'});
+
+        const user = {
+            username: 'test.one',
+            password: 'Password1',
+        } as UserProfile;
```
Comment (saturninoabril) [inline e2e/cypress/tests/integration/enterprise/ldap_group/invite_bot_spec.ts:56]: I'd suggest to use [the one we have under `Cypress` namespace](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/tests/support/index.d.ts#L31) so we import only in one place.
```suggestion
        } as Cypress.UserProfile;
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @cyberbuff! Overall looks good except for couple of requests.

File: e2e/cypress/tests/integration/enterprise/ldap_group/channel_modes_spec.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 // Stage: @prod
 // Group: @enterprise @ldap_group
+import {set} from 'lodash';
```
[the remark this answers is not in the document]
Comment (cyberbuff) [reply e2e/cypress/tests/integration/enterprise/ldap_group/channel_modes_spec.ts]: Changed all the instances.

[the remark this answers is not in the document]
Comment (cyberbuff) [reply e2e/cypress/tests/integration/enterprise/ldap_group/channel_modes_spec.ts]: Thank you 😄

File: e2e/cypress/tests/integration/enterprise/ldap_group/invite_bot_spec.ts (hunk 1 above)
Comment (cyberbuff) [reply e2e/cypress/tests/integration/enterprise/ldap_group/invite_bot_spec.ts]: Done 👍

File: e2e/cypress/tests/integration/enterprise/ldap_group/channel_modes_spec.ts [hunk 3]
```diff
@@ -18,7 +19,10 @@ describe('LDAP Group Sync - Test channel public/private toggle', () => {
         cy.apiRequireLicenseForFeature('LDAPGroups');
 
         // Enable LDAP and LDAP group sync
-        cy.apiUpdateConfig({LdapSettings: {Enable: true}});
+        cy.apiGetConfig().then(({config}) => {
+            set(config, 'LdapSettings.Enable', true);
+            cy.apiUpdateConfig(config);
+        });
```
Comment (saturninoabril) [inline e2e/cypress/tests/integration/enterprise/ldap_group/channel_modes_spec.ts:25]: Sorry. I just submitted a PR to revert the use of `lodash.set` in favor of [`DeepPartial` as in `apiUpdateConfig(newConfig: DeepPartial<AdminConfig>)`](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/tests/support/api/system.d.ts#L100). See https://github.com/mattermost/mattermost-webapp/pull/11362. Use of DeepPartial is already on master. Please update accordingly. Thanks!

**Review by saturninoabril — CHANGES_REQUESTED** [review]

File: e2e/cypress/tests/integration/enterprise/ldap_group/channel_modes_spec.ts (hunk 3 above)
Comment (cyberbuff) [reply e2e/cypress/tests/integration/enterprise/ldap_group/channel_modes_spec.ts:25]: Sure. Will do that

**Review by saturninoabril — APPROVED** [review]
Thanks @cyberbuff!

**Pull request merged**